packages/adapters/ethers/src/client.ts

```diff
@@ -1,114 +1,34 @@
+import type { CaipNetwork } from '@reown/appkit-common'
 import { ConstantsUtil as CommonConstantsUtil } from '@reown/appkit-common'
-import {
-  CaipNetworksUtil,
-  ConstantsUtil,
-  ErrorUtil,
-  HelpersUtil,
-  PresetsUtil
-} from '@reown/appkit-utils'
+import { type CombinedProvider, type ConnectorType, type Provider } from '@reown/appkit-core'
+import { ConstantsUtil, PresetsUtil } from '@reown/appkit-utils'
+import { EthersHelpersUtil, type ProviderType } from '@reown/appkit-utils/ethers'
+import { WcConstantsUtil, WcHelpersUtil, type AppKitOptions } from '@reown/appkit'
 import UniversalProvider from '@walletconnect/universal-provider'
-import type { ConnectionControllerClient, NetworkControllerClient } from '@reown/appkit-core'
-import { WcConstantsUtil } from '@reown/appkit'
-import { EthersMethods } from './utils/EthersMethods.js'
 import { formatEther, InfuraProvider, JsonRpcProvider } from 'ethers'
-import type { PublicStateControllerState } from '@reown/appkit-core'
-import { ProviderUtil, type ProviderIdType } from '@reown/appkit/store'
 import { CoinbaseWalletSDK, type ProviderInterface } from '@coinbase/wallet-sdk'
-import { W3mFrameProviderSingleton } from '@reown/appkit/auth-provider'
-
-// -- Types ---------------------------------------------------------------------
-export interface AdapterOptions {
-  ethersConfig: ProviderType
-  defaultCaipNetwork?: CaipNetwork
-}
-
-type CoinbaseProviderError = {
-  code: number
-  message: string
-  data: string | undefined
-}
-
-interface ExternalProvider extends Provider {
-  accounts: string[]
-}
-
-declare global {
-  interface Window {
-    ethereum?: Record<string, unknown>
-  }
-}
+import type { W3mFrameProvider } from '@reown/appkit-wallet'
+import { EthersMethods } from './utils/EthersMethods.js'
 
 interface Info {
   uuid: string
   name: string
   icon: string
   rdns: string
 }
-
 export interface EIP6963ProviderDetail {
   info: Info
   provider: Provider
 }
 
-export class EthersAdapter {
-  private appKit: AppKit | undefined = undefined
+export class EthersAdapter extends AdapterBlueprint {
+  private ethersConfig?: ProviderType
 
-  private EIP6963Providers: EIP6963ProviderDetail[] = []
-
-  private ethersConfig?: AdapterOptions['ethersConfig']
-
-  private authProvider?: W3mFrameProvider
-
-  // -- Public variables --------------------------------------------------------
-  public options: AppKitOptions | undefined = undefined
-
-  public caipNetworks: CaipNetwork[] = []
-
-  public chainNamespace: ChainNamespace = CommonConstantsUtil.CHAIN.EVM
-
-  public networkControllerClient?: NetworkControllerClient
-
-  public connectionControllerClient?: ConnectionControllerClient
-
-  public siweControllerClient = this.options?.siweConfig
-
-  public tokens = HelpersUtil.getCaipTokens(this.options?.tokens)
-
-  public defaultCaipNetwork: CaipNetwork | undefined = undefined
-
-  public adapterType: AdapterType = 'ethers'
+  constructor() {
+    super({})
```

Comment:
is the empty object required?

Reply:
It is a derived class, so the constructor needs a super call